src/app/core/services/modal.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ModalService {
+  isVisible = new BehaviorSubject(false);
+  isVisible$ = this.isVisible.asObservable();
+
+  open() {
+    this.isVisible.next(true);
+    document.body.style.overflow = 'hidden';
```

Comment:
A class would be more semantic and prevent override values that someone else might set on that object (super edge case but worth to mention)